packages/core/contracts/financial-templates/contract-for-difference/ContractForDifference.sol

```diff
@@ -119,15 +119,17 @@ contract ContractForDifference is Testable, Lockable {
     ) Testable(_timerAddress) {
         finder = _finderAddress;
         require(_expirationTimestamp > getCurrentTime());
-        require(_getIdentifierWhitelist().isIdentifierSupported(_priceIdentifier));
-        require(address(_financialProductLibraryAddress) != address(0));
+        require(_getIdentifierWhitelist().isIdentifierSupported(_priceIdentifier), "Identifier not registered");
+        require(address(_getOptimisticOracle()) != address(0), "Invalid finder");
```

Comment:
Should the comment be "Invalid OptimisticOracle"?

Reply:
nope. this tests that the _finder_ is set correctly as the call `_getOptimisticOracle` queries the `finder` for the `OracleInterfaces.OptimisticOracle`. This call will error if the finder is not set correctly.